napari/layers/base/base.py

```diff
@@ -145,21 +145,30 @@ def __init__(
         if translate is None:
             translate = [0] * ndim
 
-        # Create main transform mapping data to a world-like coordinate
-        self._transform = ScaleTranslate(scale, translate)
-        # Create additional transform mapping viewed data to data coordinate
-        # This is useful for cases where the displayed data is not the base
-        # data given by the user. In such cases, we need to transform the
-        # coordinates from the displayed data space to the original input data
-        # space. One example of such a situation is viewing a lower resolution
-        # level of an input pyramid, where _transform contains the transform
-        # only from the base level to world coordinates, another is when an
-        # image is larger than the maximum allowed texture size and has been
-        # downsampled
-        self._transform_view = ScaleTranslate(np.ones(ndim), np.zeros(ndim))
-        # Create additional transform mapping world-coordinates into a grid
-        # for looking at layers side-by-side
-        self._transform_grid = ScaleTranslate(np.ones(ndim), np.zeros(ndim))
+        # Create a transform chain consisting of three transforms:
+        # 1. `view2data`: An initial transform mapping the actually viewed data
+        #   to the data coordinates. This is useful for cases where the
+        #   displayed data is not the base data given by the user. In such
+        #   cases, we need to  transform the coordinates from the displayed
+        #   data space to the original input data space. One example of such a
+        #   situation is viewing a lower resolution level of an input pyramid,
+        #   another is when an image is larger than the maximum allowed texture
+        #   size and has been downsampled so it can be viewed.
+        # 2. `data2world`: The main transform mapping data to a world-like
+        #   coordinate.
+        # 3. `world2grid`: An additional transform mapping world-coordinates
+        #   into a grid for looking at layers side-by-side.
+        self._transforms = TransformChain(
```

Comment:
This is probably the most important point, but I've been stuck on what/how to say this.

I assume that transforms arranged in a chain will "flow" in a direction. Eg: from the data to what you see on screen, or vice versa. This section is confusing for me because it seems like each end of the transform chain (`view` and `grid`) relate to the canvas  you see on your monitor. The direction of the "flow" isn't as obvious as it should be.

It's possible I am confused because I'm mistaken and `grid` is not about the canvas view & instead is spatial. Is this the case? But if `grid` is spatial & changing the underlying position of the data in the world space, that would cause problems eg: picking points/shapes.

I mean, I guess that's exactly the behaviour we have now https://github.com/napari/napari/issues/860, and you do say "This PR is not designed to change any functionality" so maybe this is intended to be addressed in a later PR?

Reply:
I have to say I'm similarly confused... Why does the chain start at `view` and not `data`? (`grid` I get, it's spreading the world out on the canvas if desired. But, view?

Reply:
Ok I've re-read the paragraph and it's a little bit clearer... I don't think `view` is the best name here... `processed_data2data`? A bit too verbose...? Something we could brainstorm.

Reply:
ok, yeah - this is showing how bad a word `view` is - maybe `processed` is better, I don't love `processed_data`, we can discuss. I need to thing more, but i totally get how this is confusing and should change

Reply:
Maybe `downsampled_data`?

Reply:
love it. Given that the way this will actually appear is `downsampled_data2data` could we get away with just `downsampled2data` or is that less clear?

Reply:
@jni suggested `tile2data` I will rename and add comment / explanation

Reply:
switched to `tile2data`, explanation updated